src/pages/createCostModelWizard/markup.tsx

```diff
@@ -15,51 +15,79 @@ import React from 'react';
-                {t('cost_models_wizard.markup.sub_title')}
-              </Text>
-            </TextContent>
-          </StackItem>
-          <StackItem>
-            <Form>
-              <FormGroup
-                label={t('cost_models_wizard.markup.markup_label')}
-                fieldId="markup"
-                helperTextInvalid={t(
-                  'cost_models_wizard.markup.invalid_markup_text'
-                )}
-                isValid={isValidMarkup}
-              >
-                <InputGroup style={{ width: '150px' }}>
-                  <TextInput
-                    type="text"
-                    id="markup"
-                    name="markup"
-                    value={markup}
-                    onChange={onMarkupChange}
-                    isValid={isValidMarkup}
-                    placeholder={'0'}
-                  />
-                  <InputGroupText style={{ borderLeft: '0' }}>%</InputGroupText>
-                </InputGroup>
-              </FormGroup>
-            </Form>
-          </StackItem>
-        </Stack>
-      );
-    }}
-  </CostModelContext.Consumer>
-);
+interface MarkupValidationState {
+  isValid: boolean;
+}
+
+class Markup extends React.Component<
+  InjectedTranslateProps,
+  MarkupValidationState
+> {
+  public state = {
+    isValid: true,
+  };
+  public timer = null;
+
+  public render() {
+    const { t } = this.props;
+    const { isValid } = this.state;
+
+    return (
+      <CostModelContext.Consumer>
+        {({ onMarkupChange, markup }) => {
+          return (
+            <Stack gutter="md">
+              <StackItem>
+                <Title size="xl">{t('cost_models_wizard.markup.title')}</Title>
+              </StackItem>
+              <StackItem>
+                <TextContent>
+                  <Text component={TextVariants.h6}>
+                    {t('cost_models_wizard.markup.sub_title')}
+                  </Text>
+                </TextContent>
+              </StackItem>
+              <StackItem>
+                <Form>
+                  <FormGroup
+                    label={t('cost_models_wizard.markup.markup_label')}
+                    fieldId="markup"
+                    helperTextInvalid={t(
+                      'cost_models_wizard.markup.invalid_markup_text'
+                    )}
+                    isValid={isValid}
+                  >
+                    <InputGroup style={{ width: '150px' }}>
+                      <TextInput
+                        type="text"
+                        id="markup"
+                        name="markup"
+                        value={markup}
+                        onChange={(value: string) => {
+                          if (this.timer) {
+                            clearTimeout(this.timer);
+                          }
+                          this.timer = setTimeout(() => {
+                            this.setState({ isValid: !isNaN(Number(value)) });
+                          }, 280);
```

Comment:
Using timeouts is typically frowned upon because it's not best practice. Perhaps it would be better to perform the validation via onBlur, when the user leaves the input? 

With the onChange event, validation is performed with each key press, which isn't desirable.

Reply:
If we validate the value onChange the error message will still be shown as it does right now.
Using `onBlur` is also not perfect because if you type `minus` and focus on a different thing and then update to `-1` the error message will be shown unless you're focusing on a different element again.

Reply:
If you type minus, then leave the input, it would be expected to invalid. Regardless, my point is that this should be done without a timeout

Reply:
I see what you mean. OK - I will look at it.

Reply:
OK - I updated the code, hope you'll like it.